kong/plugins/ip-restriction/handler.lua

```diff
@@ -52,31 +53,64 @@ local function match_bin(list, binary_remote_addr)
 end
 
 
-function IpRestrictionHandler:access(conf)
+local function do_exit(status, message, is_http)
+  if is_http then
+    return kong.response.error(status, message)
+
+  else
+    local tcpsock, err = ngx.req.socket(true)
+    if err then
+      error(err)
+    end
+
+    tcpsock:send(cjson.encode({
+      status  = status,
+      message = message
+    }))
+
+    return ngx.exit()
+  end
+end
+
+
+local function handler(conf, is_http)
   local binary_remote_addr = ngx_var.binary_remote_addr
   if not binary_remote_addr then
-    return kong.response.error(403, "Cannot identify the client IP address, unix domain sockets are not supported.")
+    local status = 403
+    local message = "Cannot identify the client IP address, unix domain sockets are not supported."
+
+    do_exit(status, message, is_http)
   end
 
   local deny = conf.deny
   local allow = conf.allow
   local status = conf.status or 403
-  local message = conf.message or "Your IP address is not allowed"
+  local message = conf.message or string.format("IP address not allowed: %s", binary_remote_addr)
```

Comment:
localize `string.format`, and `binary_remote_addr` may not be formated as `%s`.

Reply:
We work on the same team.  We need this feature in 2.8.x because we can't upgrade to v3 yet.

Reply:
I'm not sure what you mean by localize.  Put the value in a local variable first?

Reply:
like `local string_format = string.format`.